core/src/main/clojure/xtdb/types.clj

```diff
@@ -836,6 +852,29 @@
              (get-elem bb start typlen idx))
            (range length)))))
 
+(defn read-binary-text-array [^bytes ba]
+  (let [^ByteBuffer bb (ByteBuffer/wrap ba)
+        ;; _payload-size (.getInt bb 0)
+        ;; _dimensions (.getInt bb 4)
+        data-offset (.getInt bb 8)
+        ;; _oid (.getInt bb 12)
+        length (.getInt bb 16)
+        get-start (fn ^long [^long header-length ^long data-offset] (+ header-length data-offset))
```

Comment:
IIRC Clojure won't be able to keep these as primitives, that only works with functions under a var, not a locally-defined function. Given they're small and used once (AFAICT?), I'd recommend inlining here?

Reply:
this is purely to circumvent reflection warnings - if i inline they come back.. we don't like them. do we?